Quadriga/src/main/java/edu/asu/spring/quadriga/web/LoginController.java

```diff
@@ -70,22 +76,39 @@ public int compare(IProject o1, IProject o2) {
         });
         int size = projectList.size() > 4 ? 4 : projectList.size();
         model.addAttribute("latestProjects", projectList.subList(0, size));
-		return "login";
+        return "login";
 
-	}
+    }
 
-	/**
+    /**
 	 * Authentication failed. User credentials mismatch causes this request.
 	 * 
 	 * @return		Redirected to login page
 	 */
 	@NoAuthorizationCheck
 	@RequestMapping(value = "/loginfailed", method = RequestMethod.GET)
-	public String loginerror(ModelMap model) {
-
-		model.addAttribute("error", "true");
+	public String loginerror(@RequestParam(value = "socialsignin", required = false) String socialSignInError,ModelMap model) {
+	    model.addAttribute("error", "true");
 		return "login";
-
 	}
 
+    @NoAuthorizationCheck
+	@RequestMapping(value = "/sociallogin", method = RequestMethod.GET)
+	    public String socialloginerror(@RequestParam(value="type", required = false) String type,ModelMap model) {
+            String socialSignInErrorMessage = "Social Login Failed. Please try again later";
+	        if(type != null){
+	            
+	            switch(type){
+	                case "1": socialSignInErrorMessage = "Account request successfully placed. The Admin will approve your request to complete the registration process.";
+	                    break;
+	                case "2": socialSignInErrorMessage = "System error. Could not complete the sign-up. Please try again later.";
+	                    break;
+	                case "3": socialSignInErrorMessage = "Admin approval of your account request is pending. Please try again later";
+	                    break;
+	            }
```

Comment:
hard-coding strings is very bad practice. hard-coding GUI string even more so. They should be coming from the messages properties file. See https://github.com/diging/quadriga/blob/9fa10b109d7f87551ac262059e05cb0dad15d598/Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/ModifyProjectController.java#L145 for an example how to do that.